src/Microsoft.OData.Core/Json/JsonValueUtils.cs

```diff
@@ -433,15 +488,44 @@ internal static string GetEscapedJsonString(string inputString)
             return builder.ToString();
         }
 
+        /// <summary>
+        /// Checks if the string contains special char and returns the find the first 
+        /// index of special char if present.
+        /// </summary>
+        /// <param name="inputString">string that might contain special characters.</param>
+        /// <param name="firstIndex">first index of the special char</param>

```

Comment:
We should indicate the expected value of firstIndex if no special characters are found. I think it may be preferable to set it to -1 as it's a constant value versus the length of the input string, which changes.